argilla/src/argilla/datasets/_export/_hub.py

```diff
@@ -0,0 +1,202 @@
+# Copyright 2024-present, Argilla, Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import os
+import warnings
+from collections import defaultdict
+from tempfile import TemporaryDirectory
+from typing import Any, Type, TYPE_CHECKING, Optional, Union
+from uuid import UUID
+
+from argilla.records._mapping import IngestedRecordMapper
+from argilla.responses import Response
+from argilla.datasets._export._disk import DiskImportExportMixin
+
+if TYPE_CHECKING:
+    from argilla import Dataset
+    from argilla import Argilla
+    from argilla import Workspace
+    from datasets import Dataset as HFDataset
+
+
+class HubImportExportMixin(DiskImportExportMixin):
+    _DEFAULT_DATASET_REPO_DIR = ".argilla"
+    _DEFAULT_CONFIGURATION_FILES = ["settings.json", "dataset.json"]
+
+    def to_hub(
+        self: "Dataset",
+        repo_id: str,
+        generate_card: Optional[bool] = True,
+        *args,
+        **kwargs,
+    ) -> None:
```

Comment:
```suggestion
    def to_hub(
        self: "Dataset",
        repo_id: str,
        *,
        generate_card: Optional[bool] = True,
        **kwargs,
    ) -> None:
```

(nit) Personal preference but I would not accept `*args` as parameter + I would force to use a keyword-argument for `generate_card`. My reasoning is that positional args are harder to maintain if you want to rename, deprecate or move them. Keyword arguments on the other hand are easier to maintain.

Also accepting `*args` to forward to `hfds.push_to_hub` do not provide additional features (kwargs is enough) but could harm maintenance if you want to add a new positional argument in the future.